server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -17,31 +17,37 @@
 
 import java.io.IOException;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.LongSupplier;
 import java.util.stream.Collectors;
 
 /**
- * FileTransferTracker keeps track of translog files uploaded to the remote translog
+ * FileTransferTracker keeps track of generational translog files uploaded to the remote translog store
  */
 public class FileTransferTracker implements FileTransferListener {
 
-    private final ConcurrentHashMap<String, TransferState> fileTransferTracker;
-    private final ShardId shardId;
+    private final Map<String, TransferState> fileTransferTracker;
+    private final Map<Long, TransferState> generationTransferTracker;
```

Comment:
Do we need both the maps? If yes, can we abstract them out instead of having 2 maps?